src/python/library/tritonclient/grpc/__init__.py

```diff
@@ -232,10 +232,14 @@ def __init__(self,
                                                  private_key=pk_bytes,
                                                  certificate_chain=cc_bytes)
             self._channel = grpc.secure_channel(url, creds, options=channel_opt)
+            self._aio_channel = grpc.aio.secure_channel(url, creds, options=channel_opt)
```

Comment:
>But still as per my reading I see gRPC advices against mixing up new and current streaming API at the same time in an application. Source: https://github.com/lidizheng/proposal/blob/grpc-python-async-api/L58-python-async-api.md#co-existence-of-new-and-current-streaming-api

Ah wasn't aware of this fact. Yes, that would be a problem if tritonclient implemented the sync and async streaming APIs and then a client application tried to use both.

>I am not comfortable for the client object to manage both the new and old gRPC python API. I would suggest adding an option in init function, that allows the client object to work with new asyncio friendly grpc python API. In this mode, client object will not use old API at all.

That sounds reasonable. Do you prefer using grpcio's asyncio APIs to the proposal @deadeyegoodwin put forward to return a future?

Reply:
Do you mean if we didn't await `ModelInfer` on line 1397? In that case it would be a coroutine object, which while still awaitable, is not a future and wouldn't allow users to use [Future methods](https://docs.python.org/3/library/asyncio-future.html) like `done()` or `result()`.